sockrus.go

```diff
@@ -19,21 +21,25 @@ type Hook struct {
 //
 // For Unix networks, the address must be a file system path.
 func NewHook(protocol, address string) (*Hook, error) {
-	conn, err := net.Dial(protocol, address)
-	if err != nil {
-		return nil, err
-	}
-	return &Hook{conn: conn}, nil
+	return &Hook{conn: nil, protocol: protocol, address: address}, nil
 }
 
 // Fire send log to the defined socket
 func (h *Hook) Fire(entry *logrus.Entry) error {
+	var err error
+	if h.conn == nil {
+		err = h.dialSock()
+		if err != nil {
+			return err
+		}
+	}
 	formatter := logrus.JSONFormatter{}
 	dataBytes, err := formatter.Format(entry)
 	if err != nil {
 		return err
 	}
 	if _, err = h.conn.Write(dataBytes); err != nil {
+		_ = h.closeSock()
```

Comment:
Please add `// #nosec` to suppress `sockrus.go:42::warning: Errors unhandled.,LOW,HIGH (gas)`.